### PR TITLE
Implemented insertion sort

### PR DESCRIPTION
**Pull request by Elliott250**

### CODE REVIEW

File: insertion_sort.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+# https://en.wikipedia.org/wiki/Insertion_sort
+def insertion_sort a
+  for i in 1..(a.size - 1)
+  x = a[i]
```
Comment (sagivo) [inline insertion_sort.rb]: i would add indentation here

Comment (Elliott250) [inline insertion_sort.rb]: Whoops.  Done.

File: insertion_sort.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+# https://en.wikipedia.org/wiki/Insertion_sort
+def insertion_sort a
+  for i in 1..(a.size - 1)
+    x = a[i]
```
Comment (sagivo) [inline insertion_sort.rb]: why do you need x?
to swap in ruby you can simply do
`a, b = b, a`

File: insertion_sort.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    j = i
+    while(j > 0 && a[j-1] >= x)
+      a[j] = a[j-1]
```
Comment (sagivo) [inline insertion_sort.rb]: i mean, you can simply do here:
`a[j], a[j-1] = a[j-1], a[j]`

File: insertion_sort.rb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+      j -= 1
+    end
+  a[j] = x
```
Comment (sagivo) [inline insertion_sort.rb]: and then you don't need this line at all

File: insertion_sort.rb (hunk 3 above)
Comment (Elliott250) [inline insertion_sort.rb]: Holding a temporary variable makes it so you only need to swap once as opposed to every iteration in the while loop which I think makes it (negligibly) faster than using ruby's swap 

Here is an example
<pre>
[1,2,3,1]
</pre>

Two swaps
<pre>
[1,2,3,1]
[1,2,1,3]
[1,1,2,3]
</pre>

One swap
<pre>
[1,2,1,3]
[1,2,2,3]
[1,1,2,3]
</pre>

**Comment by sagivo** [comment]
thanks

**Pull request merged**